docs/guides/cli.md

```diff
@@ -0,0 +1,293 @@
+# The `nnbench` command-line interface (CLI)
+
+While you can always use nnbench to directly run your benchmarks in your Python code, for example as part of a workflow, there is also the option of running benchmarks from the command line.
+This way of using nnbench is especially useful for integrating into a machine learning pipeline as part of a continuous training/delivery scenario.
+
+## General options
+
+The `nnbench` CLI has the following top-level options:
+
+```commandline
+$ nnbench
+usage: nnbench [-h] [--version] [--log-level <level>]  ...
+
+options:
+  -h, --help           show this help message and exit
+  --version            show program's version number and exit
+  --log-level <level>  Log level to use for the nnbench package, defaults to NOTSET (no logging).
+
+Available commands:
+  
+    run                Run a benchmark workload.
+    compare            Compare results from multiple benchmark runs.
+```
+
+Supported log levels are `"DEBUG", "INFO", "WARNING", "ERROR"`, and `"CRITICAL"`.
+
+## Running benchmark workloads on the command line
+
+This is the responsibility of the `nnbench run` subcommand.
+
+```commandline
+$ nnbench run -h                                                                         
+usage: nnbench run [-h] [--context <key=value>] [-t <tag>] [-o <file>] [<benchmarks>]
+
+positional arguments:
+  <benchmarks>          Python file or directory of files containing benchmarks to run.
+
+options:
+  -h, --help            show this help message and exit
+  --context <key=value>
+                        Additional context values giving information about the benchmark run.
+  -t, --tag <tag>       Only run benchmarks marked with one or more given tag(s).
+  -o, --output-file <file>
+                        File or stream to write results to, defaults to stdout.
+```
+
+So to run a benchmark workload contained in a single `benchmarks.py` file, you would run `nnbench run benchmarks.py`.
+For tips on how to structure and annotate your benchmarks, refer to the [organization](organization.md) guide.
+
+For injecting context values on the command line, you need to give the key-value pair explicitly by passing the `--context` switch.
```

Comment:
I think it makes sense to also refer to the custom implementation that comes later. Otherwise me might loose someone looking for this information and - after having found this here - not reading further in the guide.